### ISSUE TITLE
Cannot combine `FULLTEXT` and `SPATIAL` in the same query

### ISSUE THREAD

**Issue by phpnode** [orientechnologies/orientdb-lucene#41]
The following query doesn't work for me:
```sql
select *,$score, $distance as dist from Thing where [lat,lon,$spatial] NEAR [53.95763, -1.08271,{"maxDistance":20}] AND [name,description] LUCENE "(name:cafe)"
```
Only the spatial part is applied, the full text search is ignored completely. If I change the order:
```sql
select *,$score, $distance as dist from Thing where [name,description] LUCENE "(name:cafe)" and [lat,lon,$spatial] NEAR [53.95763, -1.08271,{"maxDistance":20}]
```
Then the behaviour is reversed, and only the full text search is applied.
This is the case even when separating into subselects:
```sql
select *,$score from (
  select *,$distance as dist from Thing where [lat,lon,$spatial] NEAR [53.95763, -1.08271,{"maxDistance":20}]
) where [name,description] LUCENE "(name:cafe)"
```

This really limits the usefulness of the lucene indexes and means I have to resort to the old `distance()` in the meantime.

**Comment by poindextrose** [comment]
I agree this would be a nice enhancement. I'm using a workaround with `intersect()` for now. For example:

```SQL
select expand($1)
let $2 = (select *,$distance as dist from Thing where [lat,lon,$spatial] NEAR [53.95763, -1.08271,{"maxDistance":20}]),
$3 = (select *,$score from Thing where [name,description] LUCENE "(name:cafe)"),
$1 = intersect($2, $3)
```

**Comment by phpnode** [comment]
@lvca @maggiolo00 any updates on this? it's pretty vital

**Comment by arisAlexis** [comment]
I believe a filter function where you could first perform a query on distance and then apply a lucene query or any other would be nice, so the lucene query doesn't scan the whole database.

**Comment by lvca** [comment]
Hey guys, the @poindextrose's workaround works and avoid full scan.

**Comment by maggiolo00** [comment]
Fixed in 2.1.x. 
Now you can combine the two lucene conditions. The engine is till not fixed and one index will be used.
But the other condition will be correctly evaluated with the new memory index implementation.
